File: bindings/lua/Polycode/ShaderRenderTarget.lua
class "ShaderRenderTarget"

ShaderRenderTarget.SIZE_MODE_PIXELS = 0
ShaderRenderTarget.SIZE_MODE_NORMALIZED = 1

function ShaderRenderTarget:__getvar(name)
	if name == "id" then
		return Polycode.ShaderRenderTarget_get_id(self.__ptr)
	elseif name == "width" then
		return Polycode.ShaderRenderTarget_get_width(self.__ptr)
	elseif name == "height" then
		return Polycode.ShaderRenderTarget_get_height(self.__ptr)
	elseif name == "sizeMode" then
		return Polycode.ShaderRenderTarget_get_sizeMode(self.__ptr)
	elseif name == "normalizedWidth" then
		return Polycode.ShaderRenderTarget_get_normalizedWidth(self.__ptr)
	elseif name == "normalizedHeight" then
		return Polycode.ShaderRenderTarget_get_normalizedHeight(self.__ptr)
	end
end

function ShaderRenderTarget:__setvar(name,value)
	if name == "id" then
		Polycode.ShaderRenderTarget_set_id(self.__ptr, value)
		return true
	elseif name == "width" then
		Polycode.ShaderRenderTarget_set_width(self.__ptr, value)
		return true
	elseif name == "height" then
		Polycode.ShaderRenderTarget_set_height(self.__ptr, value)
		return true
	elseif name == "sizeMode" then
		Polycode.ShaderRenderTarget_set_sizeMode(self.__ptr, value)
		return true
	elseif name == "normalizedWidth" then
		Polycode.ShaderRenderTarget_set_normalizedWidth(self.__ptr, value)
		return true
	elseif name == "normalizedHeight" then
		Polycode.ShaderRenderTarget_set_normalizedHeight(self.__ptr, value)
		return true
	end
	return false
end
function ShaderRenderTarget:ShaderRenderTarget(...)
	local arg = {...}
	for k,v in pairs(arg) do
		if type(v) == "table" then
			if v.__ptr ~= nil then
				arg[k] = v.__ptr
			end
		end
	end
	if self.__ptr == nil and arg[1] ~= "__skip_ptr__" then
		self.__ptr = Polycode.ShaderRenderTarget(unpack(arg))
	end
end

function ShaderRenderTarget:__delete()
	if self then Polycode.delete_ShaderRenderTarget(self.__ptr) end
end

File: bindings/javascript/Polycode/ShaderRenderTarget.js
function ShaderRenderTarget() {
	if(arguments[0] != "__skip_ptr__") {
		this.__ptr = Polycode.ShaderRenderTarget()
	}
	Object.defineProperties(this, {
		'id': { enumerable: true, configurable: true, get: ShaderRenderTarget.prototype.__get_id, set: ShaderRenderTarget.prototype.__set_id},
		'width': { enumerable: true, configurable: true, get: ShaderRenderTarget.prototype.__get_width, set: ShaderRenderTarget.prototype.__set_width},
		'height': { enumerable: true, configurable: true, get: ShaderRenderTarget.prototype.__get_height, set: ShaderRenderTarget.prototype.__set_height},
		'sizeMode': { enumerable: true, configurable: true, get: ShaderRenderTarget.prototype.__get_sizeMode, set: ShaderRenderTarget.prototype.__set_sizeMode},
		'normalizedWidth': { enumerable: true, configurable: true, get: ShaderRenderTarget.prototype.__get_normalizedWidth, set: ShaderRenderTarget.prototype.__set_normalizedWidth},
		'normalizedHeight': { enumerable: true, configurable: true, get: ShaderRenderTarget.prototype.__get_normalizedHeight, set: ShaderRenderTarget.prototype.__set_normalizedHeight}
	})
}

ShaderRenderTarget.prototype.__get_id = function() {
	return Polycode.ShaderRenderTarget__get_id(this.__ptr)
}

ShaderRenderTarget.prototype.__set_id = function(val) {
	Polycode.ShaderRenderTarget__set_id(this.__ptr, val)
}

ShaderRenderTarget.prototype.__get_width = function() {
	return Polycode.ShaderRenderTarget__get_width(this.__ptr)
}

ShaderRenderTarget.prototype.__set_width = function(val) {
	Polycode.ShaderRenderTarget__set_width(this.__ptr, val)
}

ShaderRenderTarget.prototype.__get_height = function() {
	return Polycode.ShaderRenderTarget__get_height(this.__ptr)
}

ShaderRenderTarget.prototype.__set_height = function(val) {
	Polycode.ShaderRenderTarget__set_height(this.__ptr, val)
}

ShaderRenderTarget.prototype.__get_sizeMode = function() {
	return Polycode.ShaderRenderTarget__get_sizeMode(this.__ptr)
}

ShaderRenderTarget.prototype.__set_sizeMode = function(val) {
	Polycode.ShaderRenderTarget__set_sizeMode(this.__ptr, val)
}

ShaderRenderTarget.prototype.__get_normalizedWidth = function() {
	return Polycode.ShaderRenderTarget__get_normalizedWidth(this.__ptr)
}

ShaderRenderTarget.prototype.__set_normalizedWidth = function(val) {
	Polycode.ShaderRenderTarget__set_normalizedWidth(this.__ptr, val)
}

ShaderRenderTarget.prototype.__get_normalizedHeight = function() {
	return Polycode.ShaderRenderTarget__get_normalizedHeight(this.__ptr)
}

ShaderRenderTarget.prototype.__set_normalizedHeight = function(val) {
	Polycode.ShaderRenderTarget__set_normalizedHeight(this.__ptr, val)
}

Duktape.fin(ShaderRenderTarget.prototype, function (x) {
	if (x === ShaderRenderTarget.prototype) {
		return;
	}
	Polycode.ShaderRenderTarget__delete(x.__ptr)
})
